test/test_variables.py

```diff
@@ -108,6 +108,51 @@ def check_request_uri(req_uri):
     check_request_uri('/4%2A')
     check_request_uri('/9?q#a')
 
+    # rewrite set $request_uri before proxy
+
+    assert 'success' in client.conf(
+        {
+            "listeners": {
+                "*:8080": {"pass": "routes/a"},
+                "[::1]:8081": {"pass": "routes/b"},
+            },
+            "routes": {
+                "a": [
+                    {
+                        "action": {
+                            "rewrite": "/foo",
```

Comment:
I feel it's necessary to test URI like `/foo%2Ffoo. For example:
```
"rewrite": "/foo%2Ffoo",
```

Reply:
Added